sycl/include/CL/sycl/detail/pi.h

```diff
@@ -542,21 +544,21 @@ pi_result piProgramCreate(
   size_t        length,
   pi_program *  res_program);
 
-pi_program piclProgramCreateWithBinary( // TODO: change to return pi_result
+pi_result piclProgramCreateWithSource(
+  pi_context        context,
+  pi_uint32         count,
+  const char **     strings,
+  const size_t *    lengths,
+  pi_program *      ret_program);
```

Comment:
I think the functions' order had better be preserved

Reply:
And so is the case in pi_opencl.cpp